SeaSharpe_CVGS/SeaSharpe_CVGS/Controllers/OrderController.cs

```diff
@@ -1,327 +1,329 @@
+            }
+
+            //get orders for member (not cart)
+            var exists = db.Orders.Where(m => m.Member.Id == member.Id).Where(d => d.OrderPlacementDate != null).Any();
+
+            if (!exists)
+            {
+                //empty cart
+                TempData["message"] = "No order history";
+                return View(Enumerable.Empty<OrderHistoryViewModel>().ToList());
+            }
+
+            //get all gameIds for order Id
+            IEnumerable<int> gameIds = db.OrderItems.
+                Where(o => o.Order.Member.Id == member.Id && o.Order.OrderPlacementDate != null)
+                .Select(i => i.GameId);
+
+            IEnumerable<Order> orderIds = db.Orders.Where(m => m.Member.Id == member.Id && m.OrderPlacementDate != null).ToList();
+
+            DateTime orderDate;
+            DateTime shipDate;
+            string gameName;
+            string platformName;
+            decimal pricePaid;
+
+            List<OrderHistoryViewModel> userOrderHistory = new List<OrderHistoryViewModel>();
+            OrderHistoryViewModel ohvm;
+
+            foreach (var item in orderIds)
+            {
+                orderDate = (DateTime)item.OrderPlacementDate;
+                if (item.ShipDate != null)
+                {
+                    shipDate = (DateTime)item.ShipDate;
+                }
+                else
+                {
+                    shipDate = DateTime.Parse("1900-01-01 00:00:00");
+                }
+
+
+                IEnumerable<OrderItem> orderItemIds = db.OrderItems.Where(oi => oi.OrderId == item.Id).ToList();
+                foreach (var orderItem in orderItemIds)
+                {
+
+                    gameName = db.Games.Where(g => g.Id == orderItem.GameId).Select(ga => ga.Name).First();
+                    platformName = db.Games.Include(p => p.Platform).Where(g => g.Id == orderItem.GameId).Select(ga => ga.Platform.Name).First().ToString();
+                    pricePaid = orderItem.SalePrice;
+                    ohvm = new OrderHistoryViewModel(orderDate, shipDate, gameName, platformName, pricePaid);
+                    userOrderHistory.Add(ohvm);
+                }
+            }
+
+            return View(userOrderHistory);
+        }
+
+        /// <summary>
+        /// Show order items in cart order
+        /// </summary>
+        /// <param name="id">member id</param>
+        /// <returns>Cart view</returns>
+        public ActionResult Cart(int? id)
+        {
+            //validate user role
+            string userId = CurrentUser.Id;
+            var member = db.Members.FirstOrDefault(m => m.User.Id == userId);
+            if (member == null)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            //validate that the member has a cart
+            var exists = db.Orders.Where(m => m.Member.Id == member.Id).Where(d => d.OrderPlacementDate == null).Any();

```

Comment:
@dHofstede The where condition on orders is using m and then d? Why not use "o" and "o"? Also for future reference you can replace the last "where" with an "any" and put the where condition in it. 

`var exists = db.Orders.Where(o => o.Member.Id == member.Id).Any(o => o.OrderPlacementDate == null);`

Not saying you should change it, just an observation. Obviously neither makes any functional difference.